### ISSUE TITLE
Multi-key mapping overrides stopped working in 0.32 

### ISSUE THREAD

**Issue by gregorias** [kovidgoyal/kitty#7091]
**Describe the bug**

Multi-key mappings that override an existing mapping (e.g., `ctrl+shift+g>u`) don’t work.

**To Reproduce**

1. Put `map ctrl+shift+g>u   kitten hints --type url   --program @` into `kitty.conf`.
2. Load the new config.
3. Press `ctrl+shift+g>u`.
4. A pager will open just after `ctrl+shift+g` (expected URL hints).

**Environment details**

```
kitty 0.32.1 (08d88af2fb) created by Kovid Goyal
Darwin Grzegorzs-MBP.home 23.3.0 Darwin Kernel Version 23.3.0: Wed Dec 20 21:30:59 PST 2023; root:xnu-10002.81.5~7/RELEASE_ARM64_T6030 arm64
ProductName:		macOS ProductVersion:		14.3 BuildVersion:		23D56
Frozen: True
Paths:
  kitty: /Applications/kitty.app/Contents/MacOS/kitty
  base dir: /Applications/kitty.app/Contents/Resources/kitty
  extensions dir: /Applications/kitty.app/Contents/Resources/Python/lib/kitty-extensions
  system shell: /opt/homebrew/bin/fish
Loaded config files:
  /Users/grzesiek/.config/kitty/kitty.conf

Config options different from defaults:
Removed shortcuts:
	cmd+, →  edit_config_file
	cmd+enter →  new_window
	cmd+h →  hide_macos_app
	cmd+k →  clear_terminal to_cursor active
	cmd+m →  minimize_macos_window
	cmd+n →  new_os_window
	cmd+q →  quit
	cmd+t →  new_tab
	cmd+w →  close_tab
	ctrl+cmd+, →  load_config_file
	ctrl+cmd+f →  toggle_fullscreen
	opt+cmd+h →  hide_macos_other_apps
	opt+cmd+r →  clear_terminal reset active
	opt+cmd+s →  toggle_macos_secure_keyboard_entry
	shift+cmd+/ →  open_url https://sw.kovidgoyal.net/kitty/
	shift+cmd+[ →  previous_tab
	shift+cmd+] →  next_tab
	shift+cmd+d →  close_window
	shift+cmd+w →  close_os_window
Changed shortcuts:
	kitty_mod+g →  kitten hints --type url   --program @, show_last_command_output

Important environment variables seen by the kitty process:
	PATH                                /Applications/kitty.app/Contents/MacOS:/usr/bin:/bin:/usr/sbin:/sbin
	LANG                                en_US.UTF-8
	SHELL                               /opt/homebrew/bin/fish
	USER                                grzesiek
```

**Additional context**

Kitty used to work with this mapping before 0.32. There’s a relevant issue #7016 that was fixed in 0.32.1, but that issue only mentions single-key mappings. Multi-key mappings still don’t work in 0.32.1.

**Comment by kovidgoyal** [comment]
Works for me in master, use a nightly build or build from source to
verify.

**Comment by gregorias** [comment]
Indeed, it works with master.

**Comment by gregorias** [comment]
For documentation’s sake, the issue was addressed in 8c50632a10f714e6d12054ac137991078b648c6e.
